### ISSUE TITLE
Download Now | button not working

### ISSUE THREAD

**Issue by FlanniganColum** [rconfig/rconfig#149]
**Describe the bug**
When i configure a host and select the "Download Now" button in the UI the jobs just go into a "Pending" queue in the horizon component and does not progress. I do see the below error in the laravel logs but not sure if this is connection to it failing. It is a new environment spun up to review the product, i can confirm the device connected is working as when i do a CLI download config it works no problem and when I have a scheduled task it also works. It seems to be just for the "Download Now" function. 

```
[2024-12-16 14:00:05] production.ERROR: Symfony\Component\Mime\Address::__construct(): Argument #1 ($address) must be of type string, null given, called in /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Mail/Message.php on line 58 {"exception":"[object] (TypeError(code: 0): Symfony\\Component\\Mime\\Address::__construct(): Argument #1 ($address) must be of type string, null given, called in /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Mail/Message.php on line 58 at /var/www/html/rconfig/vendor/symfony/mime/Address.php:42)
[stacktrace]
#0 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Mail/Message.php(58): Symfony\\Component\\Mime\\Address->__construct()
#1 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/Channels/MailChannel.php(241): Illuminate\\Mail\\Message->from()
#2 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/Channels/MailChannel.php(214): Illuminate\\Notifications\\Channels\\MailChannel->addSender()
#3 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/Channels/MailChannel.php(176): Illuminate\\Notifications\\Channels\\MailChannel->addressMessage()
#4 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/Channels/MailChannel.php(84): Illuminate\\Notifications\\Channels\\MailChannel->buildMessage()
#5 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Mail/Mailer.php(316): Illuminate\\Notifications\\Channels\\MailChannel->Illuminate\\Notifications\\Channels\\{closure}()
#6 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/Channels/MailChannel.php(66): Illuminate\\Mail\\Mailer->send()
#7 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(148): Illuminate\\Notifications\\Channels\\MailChannel->send()
#8 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(106): Illuminate\\Notifications\\NotificationSender->sendToNotifiable()
#9 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Support/Traits/Localizable.php(19): Illuminate\\Notifications\\NotificationSender->Illuminate\\Notifications\\{closure}()
#10 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(101): Illuminate\\Notifications\\NotificationSender->withLocale()
#11 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/NotificationSender.php(79): Illuminate\\Notifications\\NotificationSender->sendNow()
#12 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/ChannelManager.php(39): Illuminate\\Notifications\\NotificationSender->send()
#13 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Notifications/RoutesNotifications.php(18): Illuminate\\Notifications\\ChannelManager->send()
#14 /var/www/html/rconfig/vendor/spatie/laravel-health/src/Commands/RunHealthChecksCommand.php(116): Spatie\\Health\\Notifications\\Notifiable->notify()
#15 /var/www/html/rconfig/vendor/spatie/laravel-health/src/Commands/RunHealthChecksCommand.php(33): Spatie\\Health\\Commands\\RunHealthChecksCommand->sendNotification()
#16 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Spatie\\Health\\Commands\\RunHealthChecksCommand->handle()
#17 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Container/Util.php(43): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#18 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(95): Illuminate\\Container\\Util::unwrapIfClosure()
#19 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod()
#20 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Container/Container.php(690): Illuminate\\Container\\BoundMethod::call()
#21 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Console/Command.php(213): Illuminate\\Container\\Container->call()
#22 /var/www/html/rconfig/vendor/symfony/console/Command/Command.php(279): Illuminate\\Console\\Command->execute()
#23 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Symfony\\Component\\Console\\Command\\Command->run()
#24 /var/www/html/rconfig/vendor/symfony/console/Application.php(1047): Illuminate\\Console\\Command->run()
#25 /var/www/html/rconfig/vendor/symfony/console/Application.php(316): Symfony\\Component\\Console\\Application->doRunCommand()
#26 /var/www/html/rconfig/vendor/symfony/console/Application.php(167): Symfony\\Component\\Console\\Application->doRun()
#27 /var/www/html/rconfig/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(197): Symfony\\Component\\Console\\Application->run()
#28 /var/www/html/rconfig/artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle()
#29 {main}
"} 

```

**To Reproduce**
Just create any Device and click "Download Now"

**Expected behavior**
A task to download the config would run 

**Screenshots**

**rConfig Version Information (You can copy this from the rConfig dashboard):**     6.0.0-core
OS VERSION: Rocky 8.10
PHP: 8.3.14
REDIS VERSION: 5.0.3
MYSQL VERSION: 10.3.39 

**Logs info (can be retrieved from /var/www/html/rconfig6/storage/logs - try to paste relevant logs only)

**Additional context**

**Comment by stephenstack** [comment]
is this still an issue?

**Comment by FlanniganColum** [comment]
Yes i click the button and the job is just in the queue, please see the attached snippet.

![image](https://github.com/user-attachments/assets/bd0fcaa6-fbf8-404c-9d39-d0a80e067626)

**Comment by FlanniganColum** [comment]
It looks like when i have "php artisan queue:work --queue=ManualDownloadQueue" running it works, but when i run this its running as a foreground. Is there a systemd process that should be running this periodically or do i need to manually create my own?

**Comment by FlanniganColum** [comment]
As a temp workaround i created a new "ManualDownloadQueue.ini" file in the supervisord config directory and restarted it and now i have this deamon running in the background. However if you could provide me the correct fix that would be great but this will do me for now.

Below is what i used, of course it would prob need tweaking but as an initial test this is what i have running

````
; Horizon rConfig Configuration
[program:ManualDownloadQueue]
process_name=%(program_name)s
command=php /var/www/html/rconfig/artisan queue:work --queue=ManualDownloadQueue --daemon
autostart=true
autorestart=true
user=root
redirect_stderr=true
stdout_logfile=/var/www/html/rconfig/storage/logs/ManualDownloadQueue.log
stopwaitsecs=3600
````

**Comment by FlanniganColum** [comment]
Hey.
   So while doing more digging i found that if i update "config/horizon.php" file by appending "ManualDownloadQueue" to the "queue" array's that this would solve the issue without the need for the above supervisord config. can you advise which is the ideal method. 

If its the "config/horizon.php" file than can you please add this to the code base so that an amendment is not needed or is this an absolute post requirement to either have the supervisord config or modify the php code..

Below is the change i made. So it seems i can either have the config or modify to include the queue in the list to get the "DownloadNow" button to work as expected..

```
   'environments' => [
        'production' => [
            'supervisor-1' => [
                'connection' => 'redis',
                'queue' => ['default', 'downloadqueue', 'lastqueue', 'ManualDownloadQueue'],
                'balance' => 'auto',
                'processes' => 10,
                'tries' => 1,
                'timeout' => env('HORIZON_PROD_TIMEOUT', 60),
            ],
        ],

        'local' => [
            'supervisor-1' => [
                'connection' => 'redis',
                'queue' => ['default', 'downloadqueue', 'lastqueue', 'ManualDownloadQueue'],
                'balance' => 'auto',
                'processes' => 3,
                'tries' => 1,
                'timeout' => env('HORIZON_LOCAL_TIMEOUT', 60),
            ],
        ],
    ],

```

**Comment by stephenstack** [comment]
Great find, and an unfortunate bug on our part. We are working on a huge UI update coming soon in the new year, as well as other features on V6 Core, soon to be V8 Core. This issue slipped into the wrong branch. 
Nice find though. Shoot a mail over to info@rconfig.com and we'll send out something as part our new bug hunting program in the new year.

**Comment by stephenstack** [comment]
Closed with #152
